### DISCUSSION TITLE
OCSP Revocation in strict policy

### DISCUSSION THREAD

**Discussion by ARCH0911** [strongswan/strongswan#2112] [General]
Hi,
I'm encountering an issue with the OCSP strict policy while checking the revocation status of the remote server certificate. The validation process also checks the revocation status of the sub-CA certificate, even though it does not contain an OCSP responder URI. This is causing the validation to fail. I have attached the logs for reference.

Could you please confirm if StrongSwan checks for OCSP revocation for the entire chain of certificates, including sub-CA certificates, during the validation of the device certificate? If so, could you also provide instructions on how to disable this feature?
[logs.docx](https://github.com/strongswan/strongswan/files/14359373/logs.docx)

Thank you,
Arvind Chauhan

**Comment by tobiasbrunner** [comment]
Please post the log in a plaintext format.

**Reply by ARCH0911** [reply]
Please find the logs below-

received end entity cert "C=XX, O=CommScope, OU=SWROB Device PKI, OU=gNB Node IPSec Certificate, CN=commonname"
received issuer cert "C=XX, O=PKI Center, OU=Test SWROB Device PKI, CN=Test SWROB Device Sub-CA 1"
  using certificate "C=XX, O=CommScope, OU=SWROB Device PKI, OU=gNB Node IPSec Certificate, CN=commonname"
  using untrusted intermediate certificate "C=XX, O=PKI Center, OU=Test SWROB Device PKI, CN=Test SWROB Device Sub-CA 1"
checking certificate status of "C=XX, O=CommScope, OU=SWROB Device PKI, OU=gNB Node IPSec Certificate, CN=commonname"
  requesting ocsp status from 'http://ocsp.staging.pkiworks.com/OcspResponder/ocsp.svc' ...
  ocsp response correctly signed by "C=XX, O=CommScope, CN=Test CommScope Public Trust RSA OCSP Responder Certificate-01"
  ocsp response is valid: until Mar 11 06:43:29 2024
certificate status is good
  using trusted ca certificate "C=XX, O=PKI Center, OU=Test SWROB Device PKI, CN=Test SWROB Device Root CA"
checking certificate status of "C=XX, O=PKI Center, OU=Test SWROB Device PKI, CN=Test SWROB Device Sub-CA 1"
  ocsp response correctly signed by "C=XX, O=CommScope, CN=Test CommScope Public Trust RSA OCSP Responder Certificate-01"
  ocsp response contains no status on our certificate
  reached self-signed root ca with a path length of 1
authentication of 'C=XX, O=CommScope, OU=SWROB Device PKI, OU=gNB Node IPSec Certificate, CN=commonname' with RSA_EMSA_PKCS1_SHA2_256 successful
constraint check failed: RULE_CRL_VALIDATION is SKIPPED, but requires at least GOOD
selected peer config 'home3' inacceptable: constraint checking failed
no alternative config found

**Reply by tobiasbrunner** [reply]
If you enable `strict` revocation checking, valid OCSP or CRL status information has to be available for all certificates (except the root). If you don't want that, use `ifuri` so this is only enforced for certificates that contain CRL/OCSP URIs.

**Reply by ARCH0911** [reply]
Thank you for quick respond.
Will change the policy to ifuri then!!
